macaroons/store.go

```diff
@@ -0,0 +1,136 @@
+package macaroons
+
+import (
+	"crypto/rand"
+	"fmt"
+
+	"github.com/boltdb/bolt"
+)
+
+var (
+	// rootKeyBucketName is the name of the root key store bucket.
+	rootKeyBucketName = []byte("macrootkeys")
+
+	// defaultRootKeyID is the ID of the default root key. The first is
+	// just 0, to emulate the memory storage that comes with bakery.
+	// TODO(aakselrod): Add support for key rotation.
+	defaultRootKeyID = "0"
+
+	// macaroonBucketName is the name of the macaroon store bucket.
+	macaroonBucketName = []byte("macaroons")
+)
+
+// RootKeyStorage implements the bakery.RootKeyStorage interface.
+type RootKeyStorage struct {
+	*bolt.DB
+}
+
+// NewRootKeyStorage creates a RootKeyStorage instance.
+// TODO(aakselrod): Add support for encryption of data with passphrase.
+func NewRootKeyStorage(db *bolt.DB) (*RootKeyStorage, error) {
+	// If the store's bucket doesn't exist, create it.
+	err := db.Update(func(tx *bolt.Tx) error {
+		_, err := tx.CreateBucketIfNotExists(rootKeyBucketName)
+		return err
+	})
+	if err != nil {
+		return nil, err
+	}
+	// Return the DB wrapped in a RootKeyStorage object.
+	return &RootKeyStorage{db}, nil
+}
+
+// Get implements the Get method for the bakery.RootKeyStorage interface.
+func (r *RootKeyStorage) Get(id string) ([]byte, error) {
+	var rootKey []byte
+	err := r.View(func(tx *bolt.Tx) error {
+		rootKey = tx.Bucket(rootKeyBucketName).Get([]byte(id))
+		if len(rootKey) == 0 {
+			return fmt.Errorf("root key with id %s doesn't exist",
+				id)
+		}
+		return nil
+	})
+	if err != nil {
+		return nil, err
+	}
+	return rootKey, nil
+}
+
+// RootKey implements the RootKey method for the bakery.RootKeyStorage
+// interface.
+// TODO(aakselrod): Add support for key rotation.
+func (r *RootKeyStorage) RootKey() (rootKey []byte, id string, err error) {
+	id = defaultRootKeyID
+	err = r.Update(func(tx *bolt.Tx) error {
+		ns := tx.Bucket(rootKeyBucketName)
+		rootKey = ns.Get([]byte(id))
+		// If there's no root key stored in the bucket yet, create one.
+		if len(rootKey) == 0 {
+			// Create a 24-byte root key to emulate the memory
+			// storage that comes with bakery.
+			rootKey = make([]byte, 24)
```

Comment:
Is the value `24` included anywhere within the macaroon package as a constant? If so, then it should be used here in place of the magic number. 

Also I'm wondering why the library opts to go with `24-byte` root keys rather than `32-byte` root keys...